src/web/app/pages-instructor/instructor-student-list-page/instructor-student-list-page.component.ts

```diff
@@ -67,461 +25,122 @@ interface GetCourseDetailsResponse {
-  }
-
-  get isAllPresentStudentDetailsChecked(): boolean {
-    return !!this.allPresentStudentDetails.length
-        && this.allPresentStudentDetails.length === this.allCheckedStudentDetails.length;
-  }
-
-  get isAnyPresentCoursesChecked(): boolean {
-    return !!this.allPresentCourses.length && this.allCheckedCourses.length > 0;
-  }
-
-  courseStudentListSectionDataMap: { [key: string]: StudentListSectionData[] } = {};
-
-  constructor(private route: ActivatedRoute, private router: Router, private httpRequestService: HttpRequestService,
-              private statusMessageService: StatusMessageService) { }
-
   ngOnInit(): void {
     this.route.queryParams.subscribe((queryParams: any) => {
       this.user = queryParams.user;
-      this.loadCourses(this.user);
+      this.loadCourses();
     });
   }
 
   /**
-   * Navigate to the instructor search page with query input and query type: student.
+   * Loads courses of current instructor.
    */
-  search(): void {
-    this.router.navigate(['/web/instructor/search'], { queryParams: { studentSearchkey: this.searchQuery } });
-  }
-
-  /**
-   * Function to get initial data of the instructor's courses.
-   */
-  loadCourses(user: string): void {
-    const paramMap: { [key: string]: string } = { user };
-    this.httpRequestService.get('/instructor/students/courses', paramMap)
-        .subscribe((resp: GetCourseResponse) => {
-          this.courses = resp.courses;
-
-          if (!this.courses) {
-            this.statusMessageService.showWarningMessage('You do not have any courses yet.');
-          }
+  loadCourses(): void {
+    this.httpRequestService.get('/courses', {
+      entitytype: 'instructor',
+      coursestatus: 'active',
+    }).subscribe((courses: Courses) => {
+      this.courseList = courses;
+      this.viewStudent = Array<boolean>(this.courseList.courses.length).fill(false);
 
-        }, (resp: ErrorMessageOutput) => {
-          this.statusMessageService.showErrorMessage(resp.error.message);
-        });
-  }
-
-  /**
-   * Change value of the checkbox Display Archive Courses.
-   */
-  toggleDisplayArchive(): void {
-    this.isDisplayArchive = !this.isDisplayArchive;
-    const isDisplayArchiveChecked: boolean = this.isDisplayArchive;
-    this.courses.forEach((course: Course) => {
-      if (course && course.isArchived) {
-        this.toggleCourseStateAtInput(course, isDisplayArchiveChecked);
-      }
+    }, (resp: ErrorMessageOutput) => {
+      this.statusMessageService.showErrorMessage(resp.error.message);
     });
   }
 
   /**
-   * Change states of all courses.
+   * Toggles specific card and loads students if needed
    */
-  toggleAllPresentCoursesStateAtInput(defaultState: boolean): void {
-    this.allPresentCourses.forEach((course: Course) => {
-      if (course) {
-        this.toggleCourseStateAtInput(course, defaultState);
+  toggleCard(index: number, courseId: string): void {
+    if (this.viewStudent[index]) {
```

Comment:
How about `this.viewStudent[index] = !this.viewStudent[index]`, and then load data if necessary

Reply:
Yes that is better